bentoml/_internal/models/model.py

```diff
@@ -66,25 +68,56 @@ def with_options(cls, **kwargs: t.Any) -> ModelOptions:
                 raise ValueError(f"Option {k} is not supported")
         return cls()
 
+    @staticmethod
+    def to_dict(options: ModelOptions) -> dict[str, t.Any]:
+        return dict(options)
+
 
 bentoml_cattr.register_structure_hook_func(
     lambda cls: issubclass(cls, ModelOptions), lambda d, cls: cls(**d)
 )
-bentoml_cattr.register_unstructure_hook(ModelOptions, lambda d: dict(d))
+bentoml_cattr.register_unstructure_hook(ModelOptions, ModelOptions.to_dict)
 
 
-@attr.define(repr=False, eq=False)
+@attr.define(repr=False, eq=False, init=False)
 class Model(StoreItem):
     _tag: Tag
     __fs: FS
 
     _info: ModelInfo
-    _custom_objects: t.Optional[t.Dict[str, t.Any]] = None
-
-    _flushed: bool = False
+    _custom_objects: dict[str, t.Any] | None = None
 
     _runnable: Runnable | None = None
 
+    def __init__(
+        self,
+        tag: Tag,
+        fs: FS,
+        info: ModelInfo,
+        custom_objects: dict[str, t.Any] | None = None,
+        runnable: Runnable | None = None,
```

Comment:
Is this meant for a reserved API to customize runnable for a model?

Reply:
Ah, this is meant to be the internal way to construct a model; I wanted to be able to set all the fields with a single init call.

If the `_internal` argument is not set it will error and inform the user they should use `bentoml.models.create` or `bentoml.framework.save`.